torch/distributed/pipelining/schedules.py

```diff
@@ -40,14 +42,49 @@ def __str__(self):
         }
         return str_map[self]
 
+    @staticmethod
+    def from_str(action):
+        if action == "F":
+            return _ComputationType.FORWARD
+        elif action == "B":
+            return _ComputationType.BACKWARD
+        elif action == "W":
+            return _ComputationType.WEIGHT
+        else:
+            raise RuntimeError(f"Invalid computation type {action}")
+
+
+F = _ComputationType.FORWARD
+B = _ComputationType.BACKWARD
+W = _ComputationType.WEIGHT
+
+
+_action_regex = re.compile(r"(\d+)([F,B,W])(\d+)")
+
 
 class _Action(NamedTuple):
     computation_type: _ComputationType
     microbatch_index: int
     stage_index: int
 
     def __repr__(self):
-        return f"{self.computation_type}{self.microbatch_index}_s{self.stage_index}"
+        return f"{self.stage_index}{self.computation_type}{self.microbatch_index}"
+
+    @staticmethod
+    def from_str(str):
+        """Reverse of __repr__"""
+        if match := _action_regex.match(str):
+            stage_index, computation_type, microbatch_index = match.groups()
+            return _Action(
+                _ComputationType.from_str(computation_type),
+                int(microbatch_index),
+                int(stage_index),
+            )
+        elif str == "":
+            return None
+        raise RuntimeError(
+            f"Invalid action string: {str}, should be formatted as [stage][action type][microbatch] e.g. 2F0"
+        )
```

Comment:
It seems this method defines a contract about the input format. Can the contract be put upfront in the method's description? Words in the `RuntimeError` would work.

Reply:
Updated the method description